packages/perspective-viewer-d3fc/src/js/axis/crossAxis.js

```diff
@@ -70,49 +65,74 @@ export const domain = settings => {
         return _domain;
     };
 
+    _domain.settingName = (...args) => {
+        if (!args.length) {
+            return settingName;
+        }
+        settingName = args[0];
+        return _domain;
+    };
+
     return _domain;
 };
 
-export const labelFunction = settings => {
-    switch (axisType(settings)) {
+export const labelFunction = (settings, valueName = "__ROW_PATH__", settingName = "crossValues") => {
+    switch (axisType(settings, settingName)) {
         case AXIS_TYPES.none:
-            return d => d.__ROW_PATH__[0];
+            return d => d[valueName][0];
         case AXIS_TYPES.time:
-            return d => new Date(d.__ROW_PATH__[0]);
+            return d => new Date(d[valueName][0]);
         case AXIS_TYPES.linear:
-            return d => d.__ROW_PATH__[0];
+            return d => d[valueName][0];
         default:
-            return d => d.__ROW_PATH__.join("|");
+            return d => d[valueName].join("|");
     }
 };
 
-export const label = settings => settings.crossValues.map(v => v.name).join(", ");
+export const label = (settings, settingName = "crossValues") => settings[settingName].map(v => v.name).join(", ");
 
-const axisType = settings => {
-    if (settings.crossValues.length === 0) {
+const axisType = (settings, settingName = "crossValues") => {
+    if (settings[settingName].length === 0) {
         return AXIS_TYPES.none;
-    } else if (settings.crossValues.length === 1) {
-        if (settings.crossValues[0].type === "datetime") {
+    } else if (settings[settingName].length === 1) {
+        if (settings[settingName][0].type === "datetime") {
             return AXIS_TYPES.time;
         }
     }
     return AXIS_TYPES.ordinal;
 };
 
-export const styleAxis = (chart, prefix, settings) => {
-    chart[`${prefix}Label`](label(settings));
+const getMaxLengthsFromDomain = (domain, valueCount) => {
+    const splitLengths = domain.map(d => d.split("|").map(e => e.length));
 
-    const valueSize = v => v.length * 5;
-    const valueSetSize = s => s.split && s.split("|").reduce((m, v) => Math.max(m, valueSize(v)), 0);
-    const labelSize = prefix === "x" ? 25 : domain(settings)(settings.data).reduce((m, v) => Math.max(m, valueSetSize(v)), 0);
+    const maxLengths = [];
 
-    switch (axisType(settings)) {
+    for (let i = 0; i < valueCount; i++) {
+        const lengths = splitLengths.map(a => a[i]);
+        maxLengths.push(Math.max(...lengths));
+    }
+
+    return maxLengths;
+};
+
+export const styleAxis = (chart, prefix, settings, settingName = "crossValues", suppliedDomain) => {

```

Comment:
I am happy to approve this as it is for now and come back to it, but I just wanted to quickly ask whether it's possible to do:

````javascript
const suppliedDomain = chart[`${prefix}Domain`]();
````

Thus avoiding having to pass `suppliedDomain` in as a parameter, and avoiding the affects this has on the calling code.
We're likely to delegate the sizing code to another component soon anyway.

Reply:
Implemented as suggested.